jaeger_client/codecs.py

```diff
@@ -296,7 +296,7 @@ def inject(self, span_context, carrier):
             carrier[self.sampled_header] = '1'
 
     def extract(self, carrier):
-        if not isinstance(carrier, dict):
+        if not hasattr(carrier, 'items'):
```

Comment:
it doesn't seem like it'd work in py2. maybe `six` has an equivalent test function